plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/CaseOfCase.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE LambdaCase   #-}
+{-# LANGUAGE TypeFamilies #-}
+
+{- | A special case of case-of-case optimisation: transforms
+
+@
+    case ((force ifThenElse) b (constr t) (constr f)) alts
+@
+
+into
+
+@
+    force ifThenElse b (case (constr t) branches) (case (constr f) branches)
```

Comment:
do we need `force` here?

Reply:
I think so. `ifThenElse` is a polymorphic builtin needing a type argument.